frappe/utils/data.py

```diff
@@ -343,7 +343,10 @@ def rounded(num, precision=0):
 	if not precision and decimal_part == 0.5:
 		num = floor if (floor % 2 == 0) else floor + 1
 	else:
-		num = round(num)
```

Comment:
This entire function can be replaced with the in-built `decimal` module. Something like:

```python
from decimal import Decimal, ROUND_HALF_EVEN

prec = cint(precision)
places = Decimal(10) ** -prec
num = Decimal(num).quantize(places, rounding=ROUND_HALF_EVEN)
return float(num)
```

Reply:
Won't that cause a change in the behaviour of rounded?

Reply:
With the inputs as a number and a required precision value, it'll give the same output. Use tests, just in case. I did a simple console check, and it's giving the same result.

Reply:
This is how `rounded()` behaves. https://github.com/frappe/frappe/pull/7730

Reply:
This is what I get, but this agrees with the banker's rounding algorithm:

|  | Py2  | Py3 |
| ------------- | ------------- | ------------- |
| rounded(0.125, 2)  | 0.12  | 0.12  |
| rounded(12.5)  | 12.0  | 12.0  |